prow/slack/reporter/reporter.go

```diff
@@ -80,6 +80,13 @@ func (sr *slackReporter) GetName() string {
 func (sr *slackReporter) ShouldReport(pj *v1.ProwJob) bool {
 	config := sr.config(pj.Spec.Refs)
 
+	if !pj.Spec.Report {
```

Comment:
IMHO this belongs into crier and not into individual reporters, as it's a setting that must be respected by all reporters

Reply:
I hoisted the conditional into the Crier controller.